File: ruby/hamming/hamming.rb
class Hamming
  VERSION = 1
  def self.compute(one, two)
    raise ArgumentError if (one.length != two.length)
    array_one = one.chars
    array_two = two.chars
    equals = []
    array_one.each_with_index do |letter, index|
      equals << (letter == array_two[index])
    end
  equals.count(false)  
  end
end

File: javascript/hamming/hamming.js
var Hamming = function() {};

Hamming.prototype.compute = function(a, b) {
  if (a.length !== b.length) {
    throw new Error('DNA strands must be of equal length.');
  }
  if (a == b ) {
    return 0;
  }
  else {
    var aSplit = a.split(''), bSplit = b.split('');
    var equals = [];

    aSplit.forEach( function (letter, index) {
      if (letter == bSplit[index]) {
        equals.push(true);
      }
      else  {
        equals.push(false);
      }
    });
    return equals.toString().match(/false/g).length;
  }
};

module.exports = Hamming
